src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/SphKerrSchild.hpp

```diff
@@ -35,6 +35,291 @@ class not_null;
 namespace gr {
 namespace Solutions {
 
+/*!
+ * \brief Kerr black hole in Spherical Kerr-Schild coordinates
+ *
+ * ## Introduction
+ *
+ * Given a Kerr-Schild (KS) black hole system, we denote the coordinate system
+ * using \f$\{t,x,y,z\}\f$. In the transformed system, Spherical Kerr-Schild
+ * (SphKS), we denote the coordinate system using
+ * \f$\{t,\bar{x},\bar{y},\bar{z}\}\f$. Further, when considering indexed
+ * objects, we will use Greek and Latin indices with the standard convention
+ * (\f$\mu=0,1,2,3\f$ and \f$i=1,2,3\f$ respectively), but we will use a bar to
+ * denote that a given index is in reference to the SphKS coordinate system
+ * (i.e. \f$i\f$ vs. \f$\bar{\imath}\f$ and \f$\mu\f$ vs. \f$\bar{\mu}\f$).
+ *
+ * ## Spin in the z direction
+ *
+ * ### The Transformation
+ *
+ * The Boyer-Lindquist radius for KS with its spin in the \f$z\f$ direction is
+ * defined by
+ * \f{align}{
+ *    \frac{x^2 + y^2}{r^2 + a^2} + \frac{z^2}{r^2} = 1,
+ * \f}
+ * or equivalently,
+ * \f{align}{
+ *    r^2 &= \frac{1}{2}\left(x^2+y^2+z^2-a^2\right) +
+ *    \left(\frac{1}{4}\left(x^2+y^2+z^2-a^2\right)^2 + a^2z^2\right)^{1/2}.
+ * \f}
+ * The SphKS coordinates
+ * \f{align}{
+ *    \vec{\bar{x}} = x^{\bar{\imath}} = x_{\bar{\imath}} =
+ *    (\bar{x},\bar{y},\bar{z}),
+ * \f}
+ * and KS coordinates
+ * \f{align}{
+ *    \vec{x} = x^{i} = x_{i} = (x,y,z),
+ * \f}
+ * are related by
+ * \f{align}{
+ *    \left(\frac{\bar{x}}{r},\frac{\bar{y}}{r},\frac{\bar{z}}{r}\right) \equiv
+ *    \left(\frac{x}{\rho},\frac{y}{\rho},\frac{z}{r}\right),
+ * \f}
+ * where we have defined
+ * \f{align}{
+ *   \rho \equiv r^2 + a^2.
+ * \f}
+ * Therefore, we have that \f$r\f$ satisfies the equation for a sphere in SphKS
+ * \f{align}{
+ *    r^2 = \vec{\bar{x}}\cdot\vec{\bar{x}} = \bar{x}^2 + \bar{y}^2 +
+ *    \bar{z}^2.
+ * \f}
+ * It is clear to see that the SphKS radius coincides with the Boyer-Lindquist
+ * radius.
+ *
+ * ## Spin in an Arbitrary Direction
+ *
+ * Given that the remaining important quantities take forms that are easily
+ * specialized to the \f$z\f$ spin case, we instead focus on the general case
+ * for brevity.
+ *
+ * ### The Transformation
+ *
+ * The Boyer-Lindquist radius for KS with spin in an arbitrary direction is
+ * defined by
+ * \f{align}{
+ *    r^2 = \frac{1}{2}\left(\vec{x}\cdot\vec{x}-a^2\right) +
+ *    \left(\frac{1}{4}\left(\vec{x}\cdot\vec{x}-a^2\right)^2 +
+ *    \left(\vec{a}\cdot\vec{x}\right)^2\right)^{1/2}.
+ * \f}
+ * Then, defining two transformation matrices \f$Q^{\bar{\imath}}{}_{j}\f$ and
+ * \f$P^{j}{}_{\bar{\imath}}\f$ as
+ * \f{align}{
+ *    Q^{\bar{\imath}}{}_{j} &= \frac{r}{\rho}\delta^{\bar{\imath}}{}_{j} +
+ *    \frac{1}{(\rho + r)\rho}a^{\bar{\imath}}a_{j}, \\
+ *    P^{j}{}_{\bar{\imath}} &= \frac{\rho}{r}\delta^{j}{}_{\bar{\imath}} -
+ *    \frac{1}{(\rho + r)r}a^{j}a_{\bar{\imath}},
+ * \f}
+ * where the definition of \f$\rho\f$ is identical to Eq. (6), such that
```

Comment:
I _think_ you can do `\label{eq:BLAH}` and `\ref{eq:BLAH}` in doxygen too, instead of hard-coding `(6)`